lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -504,43 +504,39 @@ void SetMessageManagerHandler::service_handle_request_on_paging(
     std::function<void(Status, SmContextVoid)> response_callback) {
   // extract IMSI value from proto
   auto imsi = notif.common_context().sid().id();
+  uint32_t pdu_id = notif.rat_specific_notification().pdu_session_id();
+  SessionSearchCriteria criteria(imsi, IMSI_AND_PDUID, pdu_id);
   auto session_map = session_store_.read_sessions({imsi});
-  int count = 0;
-  auto session_update = SessionStore::get_default_session_update(session_map);
-  for (auto& session : session_map[imsi]) {
-    if (session->get_state() == INACTIVE) {
-      auto session_id = session->get_session_id();
-      SessionStateUpdateCriteria& session_uc = session_update[imsi][session_id];
-      m5g_enforcer_->m5g_move_to_active_state(session, notif, &session_uc);
-      bool update_success = session_store_.update_sessions(session_update);
-      if (!update_success) {
-        MLOG(MINFO) << "Operation aborted in  middle"
-                    << " session id: " << session->get_session_id() << imsi
-                    << " of imsi";
-        continue;
-      }
-      MLOG(MDEBUG) << "Successfully updated SessionStore "
-                   << " session id: " << session->get_session_id() << imsi
-                   << " of imsi";
-      count++;
-    }
-  }
-  if (!count) {
+  auto session_it = session_store_.find_session(session_map, criteria);
+  if (!session_it) {
     MLOG(MINFO) << " No Valid session found for IMSI: " << imsi;
     Status status(grpc::NOT_FOUND, "Sesion Not found");
     response_callback(status, SmContextVoid());
     return;
   }
-  bool update_success = session_store_.update_sessions(session_update);
-  if (!update_success) {
-    Status status(grpc::ABORTED, "update operation aborted in  middle");
-    response_callback(status, SmContextVoid());
-    return;
+  auto& session = **session_it;
+  auto session_update = SessionStore::get_default_session_update(session_map);
+  if (session->get_state() == INACTIVE) {
+    auto session_id = session->get_session_id();

```

Comment:
Let's try to use types when we know what it is for readability
```suggestion
    const std::string session_id = session->get_session_id();
```

Reply:
Modified the code as you suggested